packages/ui/src/viewer-table/InteractionPanel/TableOfContentsDropdown.tsx

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ * Copyright (c) 2025 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ *  This program and the accompanying materials are made available under the terms of
+ *  the GNU Affero General Public License v3.0. You should have received a copy of the
+ *  GNU Affero General Public License along with this program.
+ *   If not, see <http://www.gnu.org/licenses/>.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ *  EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ *  OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ *  SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ *  TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ *  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ *  IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+import type { Schema } from '@overture-stack/lectern-dictionary';
+import React from 'react';
+import Dropdown from '../../common/Dropdown/Dropdown';
+import { useThemeContext } from '../../theme/ThemeContext';
+
+export type TableOfContentsDropdownProps = {
+	schemas: Schema[];
+	onSelect: (schemaIndex: number) => void;
+};
+
+const TableOfContentsDropdown = ({ schemas, onSelect }: TableOfContentsDropdownProps) => {
+	const theme = useThemeContext();
+	const { List } = theme.icons;
+	const handleAction = (index: number) => {
+		const anchorId = `#${index}`;
+		onSelect(index);
+		window.location.hash = anchorId;
+	};
+
+	const menuItemsFromSchemas = schemas.map((schema, index) => ({
+		label: schema.name,
+		action: () => {
+			handleAction(index);
+		},
+	}));
+
```

Comment:
Re slack convo, yes this is a side effect, but it works ok here. TableOfContentsDropdown is specialized enough that having scrolling to a hash behaviour built in is fine.

If you wanted to you could move "handleAction" to the parent, and pass it on via the "onSelect" prop